### PR TITLE
ENH: Add freezability to remaining multivariate distributions

### PR DESCRIPTION
**Pull request by NamamiShanker**
#### Reference issue

gh-15617

#### What does this implement/fix?

- [x] Implement `ortho_group` freezability
- [ ] Implement `unitary_group` freezability
- [ ] Implement `random_correlation` freezability

### CODE REVIEW

File: scipy/stats/_multivariate.py
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3331]: I would say it's good to add this, but there are other issues here that I don't want to get into (should use `np.random.default_rng()`, comments on the previous two lines should not be there). 
IIRC, all multivariate distributions need this same sort of treatment. Let's do that to all of them in a separate PR and limit the scope here to just adding the frozen ortho group class. Sound good?

File: scipy/stats/tests/test_multivariate.py
Comment (mdhaber) [inline scipy/stats/tests/test_multivariate.py:1435]: Should also test `ortho_group(dim, seed=1234)` and see that the first `rvs` it produces is the same as the others.

File: scipy/stats/_multivariate.py
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3492]:
```suggestion
```
Let's put this in the same state as all the other multivariate distributions (even if it is not great).

**Review by mdhaber — COMMENTED** [review]
Looks good. Let's keep this simple, merge it, and move onto the next one. 
Please implement the change to the test, remove the `rvs` addition to the examples (we'll do it separately to all multivariate distributions), and I think we'll be set.

File: scipy/stats/_multivariate.py
Comment (NamamiShanker) [reply scipy/stats/_multivariate.py:3331]: Yeah, that sounds good. Should I remove the previous two comment lines?

File: scipy/stats/_multivariate.py
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3555]:
```suggestion
            If `seed` is None (or `np.random`), the `numpy.random.RandomState`
```

File: scipy/stats/_multivariate.py
Comment (mdhaber) [inline scipy/stats/_multivariate.py:3557]:
```suggestion
            If `seed` is an int, a new ``RandomState`` instance is used,
```

**Review by mdhaber — CHANGES_REQUESTED** [review]
Other comments: 
* Please add a `Parameters` section identical to that of `special_ortho_group_gen`. This section tells users what parameters are accepted when the class is used as a callable. (This wasn't there before because `ortho_group` was not callable, but it is needed due to the changes in this PR.)
* Not required, but would be nice: add a "See Also" section similar to `special_ortho_group`, listing at least `special_ortho_group`. I am not sure if `scipy.spatial.transform.Rotation.random` is still appropriate (because I haven't read into what this is doing).
* I would move the `__call__` method of `ortho_group` before the `_process_parameters` method just to match `special_ortho_group`, if you don't mind.

Otherwise, this looks good to me. Once these changes are made, I think I can merge.

**Comment by NamamiShanker** [comment]
>  I am not sure if scipy.spatial.transform.Rotation.random is still appropriate (because I haven't read into what this is doing).

`special_ortho_group_gen` returns a random Rotation matrix, which is relevant to `scipy.spatial.transform.Rotation.random`. I don't think `ortho_group_gen` is relevant because it just returns a random orthogonal matrix.

**Review by mdhaber — APPROVED** [review]
I'll merge when tests are green.

**Pull request merged**
